etna/articles/tests/test_models.py

```diff
@@ -0,0 +1,28 @@
+from django.test import TestCase
+
+from ..models import ArticleTag
+
+
+class TestArticleTagClean(TestCase):
+    def test_clean_preserves_existing_skos_ids(self):
+        tag = ArticleTag(name="Test", slug="test", skos_id="Preserve_Me")
+        tag.clean()
+        self.assertEqual(tag.skos_id, "Preserve_Me")
+
+    def test_clean_does_not_generate_skos_id_when_no_name_specified(self):
+        tag = ArticleTag(slug="test")
+        tag.clean()
+        self.assertEqual(tag.skos_id, "")
+
+    def test_clean_generates_skos_id_from_name_with_no_conflicts(self):
+        tag = ArticleTag(name="Test", slug="test")
+        tag.clean()
+        self.assertEqual(tag.skos_id, "Test")
+
+    def test_clean_generates_skos_id_from_name_with_conflicts(self):
+        ArticleTag.objects.create(name="Test", slug="test", skos_id="Test")
+        ArticleTag.objects.create(name="Test 2", slug="test-2", skos_id="Test_2")
+
+        tag = ArticleTag(name="Test", slug="test-3")
+        tag.clean()
+        self.assertEqual(tag.skos_id, "Test_3")
```

Comment:
Question: the UI to add article tag check unique `name` when saving, why is it needed to generate a unique `skos_id` when from `name` when it is unique - same as `slug`?

Reply:
What are your thoughts on it, @TNA-Allan? Why do you think they have unique restrictions?

Reply:
Of course it has to be since the UI allows editing for that `name` field.